src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -25,22 +24,49 @@ public class ProcessStartInfo
     /// </summary>
     public class Container : IDisposable
     {
-        private string _id;
+        private IntPtr _cs;
         private bool _killOnClose;
         private bool _killed;
 
-        internal Container(string id, bool terminateOnClose)
+        private HcsNotificationWatcher _watcher;
+
+        internal Container(string id, IntPtr computeSystem, bool terminateOnClose)
```

Comment:
I can rework it to use a static initializer.  I wasn't sure if waiting in the constructor was kosher, but the static initializer is a perfect way around that.